src/app/shared/user.service.ts

```diff
@@ -52,18 +54,20 @@ export class UserService {
         .get(this.userUrl, {headers: this.headers})
         .toPromise()
         .then(response => {
-          let userData = response.json().data as User;
+          let userData = response.json().data;
           // The reference of this.userData is
           // being used in Header
           // So updating the value like that
           this.userData.attributes = {
             fullName: userData.attributes.fullName,
-            imageURL: userData.attributes.imageURL
+            imageURL: userData.attributes.imageURL,
+            username: userData.attributes.username,
+            bio: userData.attributes.bio,
+            primaryEmail: userData.attributes.email
           };
           this.userData.id = userData.id;
+          this.profile.initDefaults(this.userData);
```

Comment:
I'm in the process of abstracting out the user.service. I don't think I should take the Profile with it. Can you have the profile listen to the broadcast service? Or find another way to set the Profile defaults?